src/reflect/value.go

```diff
@@ -992,12 +1008,29 @@ func (v Value) checkRO() {
 	}
 }
 
+// OverflowInt reports whether the int64 x cannot be represented by v's type.
+// It panics if v's Kind is not Int, Int8, Int16, Int32, or Int64.
 func (v Value) OverflowInt(x int64) bool {
-	panic("unimplemented: reflect.OverflowInt()")
+	switch v.Kind() {
+	case Int, Int8, Int16, Int32, Int64:
+		bitSize := v.typecode.Size() * 8
+		trunc := (x << (64 - bitSize)) >> (64 - bitSize)
```

Comment:
This is going to result in _terrible_ machine code on AVR but it looks correct 🤷

Reply:
This is the code from upstream :(